File: Python/1022 - TDA Racional.py
def mdc(a, b):
    while b:
        a, b = b, a % b
    return a

N = int(input())

for _ in range(N):
    entrada = input().split()
    n1 = int(entrada[0])
    d1 = int(entrada[2])
    op = entrada[3]
    n2 = int(entrada[4])
    d2 = int(entrada[6])

    if op == '+':
        num = n1 * d2 + n2 * d1
        den = d1 * d2
    elif op == '-':
        num = n1 * d2 - n2 * d1
        den = d1 * d2
    elif op == '*':
        num = n1 * n2
        den = d1 * d2
    elif op == '/':
        num = n1 * d2
        den = n2 * d1

    divisor = mdc(abs(num), abs(den))
    num_simpl = num // divisor
    den_simpl = den // divisor

    print(f"{num}/{den} = {num_simpl}/{den_simpl}")

File: Clojure/1022 - TDA Racional.clj
(defn mdc [a b]
  (if (zero? b)
    a
    (recur b (mod a b))))

(let [N (Integer/parseInt (read-line))]
  (dotimes [_ N]
    (let [entrada (clojure.string/split (read-line) #"\s+")
          n1 (Integer/parseInt (nth entrada 0))
          d1 (Integer/parseInt (nth entrada 2))
          op (nth entrada 3)
          n2 (Integer/parseInt (nth entrada 4))
          d2 (Integer/parseInt (nth entrada 6))
          [num den] (cond
                      (= op "+") [(+ (* n1 d2) (* n2 d1)) (* d1 d2)]
                      (= op "-") [(- (* n1 d2) (* n2 d1)) (* d1 d2)]
                      (= op "*") [(* n1 n2) (* d1 d2)]
                      (= op "/") [(* n1 d2) (* n2 d1)])
          divisor (mdc (Math/abs num) (Math/abs den))
          num-simpl (quot num divisor)
          den-simpl (quot den divisor)]
      (println (format "%d/%d = %d/%d" num den num-simpl den-simpl)))))